src/dbseeder/programs.py

```diff
@@ -134,116 +144,192 @@ class WqpProgram(object):
+        '''create a new WQP program
+        db - the connection string for the database to seed
+        file_location - the path on disk to find csv files to ETL
+
+        if `file_location` is None, it is assumed to be an update
+        operation
+        '''
         super(WqpProgram, self).__init__()
 
-        #: check that file_location exists wqp/results and wqp/stations
-        parent_folder = join(file_location, self.datasource)
+        self.db = db
 
-        if not isdir(parent_folder):
-            raise Exception('Pass in a location to the parent folder that contains WQP. {}'.format(parent_folder))
+        #: if file_location is None then we are updating
+        if file_location is not None:
+            #: check that file_location exists wqp/results and wqp/stations
+            parent_folder = join(file_location, self.datasource)
 
-        results_folder = join(parent_folder, 'Results')
-        stations_folder = join(parent_folder, 'Stations')
+            if not isdir(parent_folder):
+                raise Exception('Pass in a location to the parent folder that contains WQP. {}'.format(parent_folder))
 
-        if not isdir(results_folder or stations_folder):
-            raise Exception('WQP folder is missing Results or Stations child folders containing csv files.')
+            results_folder = join(parent_folder, 'Results')
+            stations_folder = join(parent_folder, 'Stations')
 
-        #: all is well set the folders for seeding later on
-        self.results_folder = results_folder
-        self.stations_folder = stations_folder
+            if not isdir(results_folder or stations_folder):
+                raise Exception('WQP folder is missing Results or Stations child folders containing csv files.')
 
-        self.db = db
+            #: all is well set the folders for seeding later on
+            self.results_folder = results_folder
+            self.stations_folder = stations_folder
 
     def seed(self):
+        if not hasattr(self, 'results_folder') or not hasattr(self, 'stations_folder'):
+            raise Exception('You must pass a file location if you are seeding. Did you want to update?')
+
+        try:
+            self._seed_by_file()
+        finally:
+            if hasattr(self, 'cursor'):
+                del self.cursor
+
+    def update(self):
+        try:
+            last_updated = self._get_most_recent_result_date()
+
+            if not last_updated:
+                raise Exception('No last updated date')
+
+            #: get new results from wqp service
+            result_url = self._format_url(self.wqp_url, 'Result', last_updated)
+            #: group them as if they were read from querycsv
+            new_results = self._group_rows_by_id(HttpClient.get_csv(result_url))
+
+            #: find the station ids from the new results that aren't in the database
+            new_station_ids = self._find_new_station_ids(new_results)
```

Comment:
I think we do need to do something here. 

Two scenarios I mapped out. 

1. You get 123 and 123_wqx where 123 is already in the databas.

I think in this situation we would insert a duplicate station.

2. You get 123 and 123_wqx where neither is already in the database.  

I think we insert the correct one. 

If we strip the wqx prior, we fix 1, but I think we then have a crap shoot for which one gets inserted for 2. 

Can you confirm this so I feel better about it?